src/contracts/examples/redstone-oev/RedstoneAdapterAtlasWrapper.sol

```diff
@@ -88,64 +100,87 @@ contract RedstoneAdapterAtlasWrapper is Ownable, MergedSinglePriceFeedAdapterWit
         uint256 minIntervalBetweenUpdates = getMinIntervalBetweenUpdates();
         if (currentBlockTimestamp < blockTimestampFromLatestUpdate + minIntervalBetweenUpdates) {
             revert MinIntervalBetweenUpdatesHasNotPassedYet(
-                currentBlockTimestamp,
-                blockTimestampFromLatestUpdate,
-                minIntervalBetweenUpdates
+                currentBlockTimestamp, blockTimestampFromLatestUpdate, minIntervalBetweenUpdates
             );
         }
     }
 
     function latestAnswer() public view virtual override returns (int256) {
         bytes32 dataFeedId = getDataFeedId();
-        uint256 baseAnswer = IAdapter(BASE_ADAPTER).getValueForDataFeed(dataFeedId);
+        (, uint256 atlasLatestBlockTimestamp) = getTimestampsFromLatestUpdate();
         uint256 atlasAnswer = getValueForDataFeed(dataFeedId);
-        if (atlasAnswer == 0) {
-            return int256(baseAnswer);
-        }
-        (uint256 baseLatestTimestamp,) = IAdapter(BASE_ADAPTER).getTimestampsFromLatestUpdate();
-        (uint256 atlasLatestTimestamp,) = getTimestampsFromLatestUpdate();
 
-        if (atlasLatestTimestamp > baseLatestTimestamp - BASE_FEED_DELAY) {
-            return int256(atlasAnswer);
+        uint80 roundId = IFeed(BASE_FEED).latestRound();
+
+        uint256 numIter;
+        for (;;) {
+            (uint256 baseValue,, uint128 baseBlockTimestamp) =
+                IAdapterWithRounds(BASE_ADAPTER).getRoundDataFromAdapter(dataFeedId, roundId);
+            if (atlasLatestBlockTimestamp >= baseBlockTimestamp) {
```

Comment:
should this be ```if (atlasLatestBlockTimestamp >= baseBlockTimestamp - BASE_FEED_DELAY) {```

Reply:
I think logically it is currently right but if the atlasLatestBlockTimestamp was only 1 block behind baseBlockTimestamp, and the delay is 4 blocks, then it would take a full iteration of this for loop to return the atlasAnswer. But if you made the change suggested above then it would return the atlasLastAnswer without a full for loop iteration?

Reply:
Done, clarified the logic.